### PR TITLE
added compatibility with Symfony3

### PR DESCRIPTION
**Pull request by Halleck45**

### CODE REVIEW

**Comment by Halleck45** [comment]
PhpMetrics is not compatible with Symfony3, due to usage of `Symfony\Component\Console\Helper\TableHelper` ([link to build](https://travis-ci.org/phpmetrics/PhpMetrics/jobs/116568158) ).

@theofidry @llaumgui Can you help us ?

**Comment by theofidry** [comment]
So we have to replace it with the [Table Component](http://symfony.com/doc/current/components/console/helpers/table.html), will take a look into that tonight.

**Comment by UFOMelkor** [comment]
IMHO we should also enhance the travis configuration to test with different dependencies (at least with `--prefer-lowest`).

File: src/Hal/Application/Formater/Details/Cli.php [hunk 1]
```diff
@@ -76,7 +75,14 @@ public function terminate(ResultCollection $collection, ResultCollection $groupe
 
         $hasOOP = null !== $total->getSum('instability');
 
-        $table = new TableHelper();
+        if(class_exists('Symfony\Component\Console\Helper\Table')) {
```
Comment (UFOMelkor) [inline src/Hal/Application/Formater/Details/Cli.php]: Is this really necessary? We require `symfony/console: ~2.5`

**Comment by Halleck45** [comment]
Yes @UFOMelkor , you're right. I've just changed the PR.

Is there a way to run tests with 2.5 **and** 3.0 with travis ?

**Comment by UFOMelkor** [comment]
It should work with environment variables and [build matrices](https://docs.travis-ci.com/user/customizing-the-build/#Build-Matrix).

Let's assume we want test the minimal requirements:

```yaml

php:
    - 5.4
    - 5.5
    - 5.6
    - 7.0
    - hhvm

matrix:
  fast_finish: true
  include:
    - php: 5.4
      env: COMPOSER_FLAGS="--prefer-lowest"

before_script:
  - composer selfupdate --quiet
  - composer update --prefer-dist --no-interaction --no-progress $COMPOSER_FLAGS
```

We can also test specific combinations. You might have a look at the [.travis configuration of the FOSUserBundle](https://github.com/FriendsOfSymfony/FOSUserBundle/blob/master/.travis.yml)

**Comment by theofidry** [comment]
@Halleck45 you may want another example with different versions: https://github.com/hautelook/AliceBundle/blob/1.x/.travis.yml

```yaml
language: php

sudo: false

php:
  - '5.5'
  - '7.0'
  - nightly
  - hhvm

matrix:
  fast_finish: true
  include:
    - php: '5.4'
      env: COMPOSER_FLAGS="--prefer-lowest"
      env: SYMFONY_VERSION=2.3.*
    - php: '5.6'
      env: SYMFONY_VERSION=2.3.*
    - php: '5.6'
      env: SYMFONY_VERSION=2.7.*
    - php: '5.6'
      env: SYMFONY_VERSION=2.8.*
    - php: '5.6'
      env: SYMFONY_VERSION=3.0.*
    - php: '5.6'
      env: SYMFONY_VERSION="3.1.*@dev"
  allow_failures:
    - php: '7.0'
    - php: nightly
    - php: hhvm
    - env: SYMFONY_VERSION="3.1.*@dev"

services:
  - mongodb
  - mysql

cache:
  directories:
    - $HOME/.composer/cache/files

before_install:
  - composer self-update
  - if [ "$SYMFONY_VERSION" != "" ]; then composer require "symfony/symfony:${SYMFONY_VERSION}" --no-update; fi;
  - echo "extension = mongo.so" >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini

install:
  - if ([ "hhvm" != "$TRAVIS_PHP_VERSION" ] && [ "nightly" != "$TRAVIS_PHP_VERSION" ]); then wget https://scrutinizer-ci.com/ocular.phar; fi;
  - composer update --prefer-dist --no-interaction $COMPOSER_FLAGS

script: make test

after_script:
  - if ([ "hhvm" != "$TRAVIS_PHP_VERSION" ] && [ "nightly" != "$TRAVIS_PHP_VERSION" ]); then php ocular.phar code-coverage:upload --format=php-clover coverage.xml; fi;

notifications:
  email: false
```

**Pull request merged**